exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Query.kt

```diff
@@ -105,6 +105,8 @@ open class Query(override var set: FieldSet, where: Op<Boolean>?) : AbstractQuer
     }
 
     override fun prepareSQL(builder: QueryBuilder): String {
+        require(set.fields.isNotEmpty()) { "Can't prepare SELECT statement without columns or expressions to retrieve" }
+
```

Comment:
The check cannot be placed directly inside the `slice()` because that would not prevent the edge case of a user creating a `Slice` instance instead of calling the method (it's odd but is possible):
```kt
MyTable.slice(emptyList()).selectAll()
// vs
Slice(MyTable, emptyList()).selectAll()
```

So the alternative location is to put the check inside `Slice`:
```kt
class Slice(override val source: ColumnSet, override val fields: List<Expression<*>>) : FieldSet {
    init {
        require(fields.isNotEmpty()) { "Can't prepare SELECT statement without columns or expressions to retrieve" }
    }
}
```